testing/integration/agent_long_test.go

```diff
@@ -0,0 +1,282 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+//go:build integration
+
+package integration
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"math"
+	"os"
+	"os/exec"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/sajari/regression"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/elastic/elastic-agent-libs/kibana"
+	"github.com/elastic/elastic-agent/pkg/control/v2/cproto"
+	atesting "github.com/elastic/elastic-agent/pkg/testing"
+	"github.com/elastic/elastic-agent/pkg/testing/define"
+	"github.com/elastic/elastic-agent/pkg/testing/tools"
+	"github.com/elastic/go-sysinfo"
+	"github.com/elastic/go-sysinfo/types"
+)
+
+type ExtendedRunner struct {
+	suite.Suite
+	info         *define.Info
+	agentFixture *atesting.Fixture
+
+	ESHost string
+}
+
+type ComponentMetrics struct {
+	Memory        MemoryMetrics  `mapstructure:"memstats"`
+	Handles       HandlesMetrics `mapstructure:"handles"`
+	UnixTimestamp int64
+}
+
+// TestComponent is used as a key in our map of component metrics
+type TestComponent struct {
+	Binary   string `mapstructure:"binary"`
+	Dataset  string `mapstructure:"dataset"`
+	ID       string `mapstructure:"id"`
+	CompType string `mapstructure:"type"`
+}
+
+type MemoryMetrics struct {
+	GcNext      uint64 `mapstructure:"gc_next"`
+	MemoryAlloc uint64 `mapstructure:"memory_alloc"`
+	MemorySys   uint64 `mapstructure:"memory_sys"`
+	MemoryTotal uint64 `mapstructure:"memory_total"`
+	RSS         uint64 `mapstructure:"rss"`
+}
+
+type HandlesMetrics struct {
+	Open  int           `mapstructure:"open"`
+	Limit HandlesLimits `mapstructure:"limit"`
+}
+
+type HandlesLimits struct {
+	Hard uint   `mapstructure:"hard"`
+	Soft uint64 `mapstructure:"soft"`
+}
+
+// MetricsSystem is used for windows handles metrics
+type MetricsSystem struct {
+	Handles HandlesMetrics `mapstructure:"handles"`
+}
+
+type processWatcher struct {
+	handle     types.Process
+	pid        int
+	name       string
+	regHandles *regression.Regression
+}
+
+func TestAgentLong(t *testing.T) {
+	info := define.Require(t, define.Requirements{
+		Group: "fleet",
+		Stack: &define.Stack{},
+		Local: false, // requires Agent installation
+		Sudo:  true,  // requires Agent installation
+		OS: []define.OS{
+			{Type: define.Linux},
+			{Type: define.Windows},
+		},
+	})
+
+	if os.Getenv("TEST_EXTENDED") == "" {
+		t.Skipf("not running extended test unless TEST_EXTENDED is set")
+	}
+
+	suite.Run(t, &ExtendedRunner{info: info})
+}
+
+func (runner *ExtendedRunner) SetupSuite() {
+	// create ~40 1MB files that will be picked up by the `/var/log/httpd/error_log*` pattern
+	cmd := exec.Command("go", "install", "-v", "github.com/mingrammer/flog@latest")
```

Comment:
Can we just import the parts of flog that we need instead of having to do this for every single test? Also see https://github.com/leehinman/spigot for something similar that we actually control.

Reply:
Ah, I knew we had our own package for that somewhere, just couldn't find it at the time.